packages/transformer/src/IModelExporter.ts

```diff
@@ -1145,6 +1220,142 @@ export class ChangedInstanceIds {
       this.handleChange(this.element, changeType, change.ECInstanceId);
   }
 
+  /**
+   * Adds the provided change to the element changes maintained by this instance of ChangedInstanceIds
+   * If the same ECInstanceId is seen multiple times, the changedInstanceIds will be modified accordingly, i.e. if an id 'x' was updated but now we see 'x' was deleted, we will remove 'x'
+   * from the set of updatedIds and add it to the set of deletedIds for the appropriate class type.
+   * @note element changes will also cause the element's model to be marked as updated in [[ChangedInstanceIds.model]], so that the element does not get skipped by the transformer.
+   * @note It is the responsibility of the caller to ensure that the provided id is, in fact an element.
+   * @note In most cases, this method does not need to be called. Its only for consumers to mimic changes as if they were found in a changeset, which should only be useful in certain cases such as the changing of filter criteria for a preexisting master branch relationship.
+   * @beta
+   */
+  public addCustomElementChange(
+    changeType: SqliteChangeOp,
+    ids: Id64Arg
+  ): void {
+    // if delete unnecessary?
+    for (const id of Id64.iterable(ids)) {
+      this.addModelToUpdated(id);
```

Comment:
To improve performance, consider querying models of all elements at once in `addModelToUpdated()`

Reply:
Switched to one query for all the elements